server/etcdmain/help.go

```diff
@@ -104,15 +104,41 @@ Clustering:
   --advertise-client-urls 'http://localhost:2379'
     List of this member's client URLs to advertise to the public.
     The client URLs advertised should be accessible to machines that talk to etcd cluster. etcd client libraries parse these URLs to connect to the cluster.
+  --enable-v2-discovery 'true'
+    Enable to bootstrap the cluster using v2 discovery. Will be deprecated in v3.7, and be decommissioned in v3.8.
   --discovery ''
     Discovery URL used to bootstrap the cluster.
+  --discovery-dial-timeout
```

Comment:
It would be way nicer if we had a single json describing any etcd client configuration: 

`--discovery-conn-json='{url: ... dial-timeout: ...}'`

this would keep the configuration up-to date with all the changes we introduce to the client without introducing a lot of custom flags.

Reply:
I think that we might want to invest into making configuration more reusable, like extracting and reusing common client configuration struct. This is however independent on the decision on introducing hybrid flag like `--discovery-conn-json` that combine both command line and JSON interface.

Reply:
I like the idea, I will try to fix it in a separate PR. Thanks @serathius 

> I think that we might want to invest into making configuration more reusable, like extracting and reusing common client configuration struct.

Reply:
SGTM. 
Even if we want to keep the flags separate and top-level, I think we should I'm at sharing the logic with: 

https://github.com/etcd-io/etcd/blob/bf0b1a7e1d5f213c2f7fd5b368aa73a79be9294c/etcdctl/ctlv3/ctl.go#L66-L80. 

such tag all the flags can be imported as: 
   
```
  DeclareV3ClientFlags(rootCmd.PersistentFlags(), &someContext.Client, "discovery", "discoveryV3: ");
```

But I'm fine with doing this in separate PR. Just it's good to align the naming.